### ISSUE TITLE
Ad not showing.

### ISSUE THREAD

**Issue by iamshrikantjha** [ammarahm-ed/react-native-admob-native-ads#39]
Hi,
I have installed the library as per instructions. But the ad view is showing a purple block only. I tried to swap ids but the result is same

![ads](https://user-images.githubusercontent.com/22591983/84398889-1d5f4280-ac1e-11ea-810d-4fe0194d1fdb.jpeg)

Here I just need some advice am I doing it right. Or what's wrong with it.

**Comment by ammarahm-ed** [comment]
Try the example app and reproduce the issue.

**Comment by KVinS** [comment]
I have the same problem...

**Comment by ammarahm-ed** [comment]
Are you using your own ad ids? If so, wait for a day or two and the ads will start showing. If your app is not published on playstore, when you publish it and it goes live. Ads will start showing. This error is not related to this library but admob server.

**Comment by ammarahm-ed** [comment]
For testing/development purposes please use the test ad & test app id from the admob website.

**Comment by KVinS** [comment]
> Are you using your own ad ids? If so, wait for a day or two and the ads will start showing. If your app is not published on playstore, when you publish it and it goes live. Ads will start showing. This error is not related to this library but admob server.

I am aware of this problem and therefore used these id =\
https://developers.google.com/admob/android/test-ads

**Comment by ammarahm-ed** [comment]
Did you run the example? Is it the same in the example app?

**Comment by KVinS** [comment]
> Did you run the example? Is it the same in the example app?

Lol It seems that this is my network firewall that blocked access to the admob.
Nonetheless:
- Video ads still not working
- Slightly strange component behavior if ads are not loaded
![2020-06-14_04-18-41](https://user-images.githubusercontent.com/3009807/84582603-b21c9880-adf6-11ea-9785-3715a52793eb.png)

**Comment by iamshrikantjha** [comment]
Hi @KVinS how did you fixed the "network firewall" issue I am using Ubuntu and running application on real device.

**Comment by ammarahm-ed** [comment]
@KVinS video ads will work when you use native-advanced-video ids. You should use `MediaView`. It will automatically load video or image based on the Ad type. You can hide the ad component from the view when ads failed to load sometimes.

**Comment by KVinS** [comment]
> Hi @KVinS how did you fixed the "network firewall" issue I am using Ubuntu and running application on real device.
@iamshrikantjha 
I went to the router and disabled ad blocking through dns.


> @KVinS video ads will work when you use native-advanced-video ids. You should use `MediaView`. It will automatically load video or image based on the Ad type. You can hide the ad component from the view when ads failed to load sometimes.
@ammarahm-ed, I understand this in theory. In the main example, a large square is responsible for this. However, the result is only a “simple” advertisement.
I could not get the video. = ( Can you provide a small ONLY video ad code that works for you?

**Comment by ammarahm-ed** [comment]
@KVinS The same code works for me. Video ads work out of the box. You just have to use native-advanced video ad ids.

**Comment by KVinS** [comment]
> @KVinS The same code works for me. Video ads work out of the box. You just have to use native-advanced video ad ids.
I use it (
'ca-app-pub-3940256099942544/1044960115'

`[Mon Jun 15 2020 04:28:57.460]  LOG      {"error": {"message": "Internal error, an invalid response was received from the ad server."}}`


```
        <NativeAdView
          onAdLoaded={_onAdLoaded}
          onAdFailedToLoad={_onAdFailedToLoad}
          style={{
            width: '100%',
            alignSelf: 'center',
            height: 412,
          }}
          adUnitID={NATIVE_AD_VIDEO_ID}>
          <View
            style={{
              height: 400,
              width: '100%',
              backgroundColor: 'white',
            }}>
            <MediaView
              style={{
                width: 400,
                height: 400 / aspectRatio,
                backgroundColor: 'gray',
              }}
            />
          </View>
        </NativeAdView>
```

**Comment by ammarahm-ed** [comment]
@KVinS No you don't. The video should play anyways. The error means that the server is not sending the ads for some reason. nothing wrong with the library. check your firewall etc or something. Use a VPN app and it will work. Your network is blocking requests still

**Comment by KVinS** [comment]
> No you don't.

And how does a user navigate through advertising without an interaction element?

**Comment by ammarahm-ed** [comment]
@KVinS I meant it is not necessary to show the button, ads work anyways. Also pressing on the icon/text etc also opens the ad and is registered as ad click.  But yea it should be added.

**Comment by KVinS** [comment]
> @KVinS I meant it is not necessary to show the button, ads work anyways. Also pressing on the icon/text etc also opens the ad and is registered as ad click. But yea it should be added.

It's work on real device (not emulator)
But crop video and with frame. Can i set auto size?
![Screenshot_2020-06-19-05-07-51-744_com howtodraw](https://user-images.githubusercontent.com/3009807/85089355-24242180-b1eb-11ea-8927-a711e49aac39.jpg)

**Comment by ammarahm-ed** [comment]
@KVinS Yes you can, use the aspectRatio value returned in `onUnifiedNativeAdLoaded`. Check the example app, it is demonstrated there how to auto size the MediaView etc.

**Comment by prasanthsd** [comment]
We are testing with test ads and on some android devices native ads are having issues and on other devices it works great. Anyone else having similar experience?

**Comment by KVinS** [comment]
@ammarahm-ed  I launched the application today and it again stopped displaying ads :D
(Simple banner)
Was there some kind of update or something?

**Comment by prasanthsd** [comment]
Google sometime wont have the icon with the ad. This isn't handled by the lib that was our issue with native ads.

**Comment by KVinS** [comment]
> Google sometime wont have the icon with the ad. This isn't handled by the lib that was our issue with native ads.

The action button works. The load event is raised.
But there is no text and image.
![2020-06-26_15-11-12](https://user-images.githubusercontent.com/3009807/85855711-5fac8600-b7bf-11ea-8be7-4013d84d3715.png)

**Comment by ammarahm-ed** [comment]
Check if you are getting ´onUnifiedNativeAdLoaded´ event called with the ad data or not. Also run adb logcat to see if there is any error exception reports logged?

**Comment by KVinS** [comment]
> There has been no update yet.
Update - in the sense that "every minute an advertising update is replaced."
```
06-26 20:21:42.966 31527 31581 I ReactNativeJS: Banner  Ad has loaded
06-26 20:22:43.722 31527 31581 I ReactNativeJS: Banner  Ad has loaded
06-26 20:23:44.384 31527 31581 I ReactNativeJS: Banner  Ad has loaded
06-26 20:24:45.022 31527 31581 I ReactNativeJS: Banner  Ad has loaded
06-26 20:25:45.665 31527 31581 I ReactNativeJS: Banner  Ad has loaded
06-26 20:26:46.344 31527 31581 I ReactNativeJS: Banner  Ad has loaded
06-26 20:27:46.936 31527 31581 I ReactNativeJS: Banner  Ad has loaded
06-26 20:28:47.534 31527 31581 I ReactNativeJS: Banner  Ad has loaded

```

> Check if you are getting ´onUnifiedNativeAdLoaded´ event called with the ad data or not.
I get an event "onAdLoaded".  I can click on the button and go to the site.
But no information is displayed.

**Comment by ammarahm-ed** [comment]
Try the example app and see if you can find the issue there also?

**Comment by KVinS** [comment]
> Try the example app and see if you can find the issue there also?

Can you send apk to me, please?

**Comment by ammarahm-ed** [comment]
You can clone this project and run the example from the example folder.

**Comment by KVinS** [comment]
> You can clone this project and run the example from the example folder.

This is difficult for me to do. I installed NPM for only one project.

**Comment by KVinS** [comment]
@ammarahm-ed Can you send an apk example with a test id?

**Comment by ammarahm-ed** [comment]
Use the ´delayAdLoading´ prop instead and see if it works for you.  Also compare your code with the example app.

**Comment by KVinS** [comment]
> Use the ´delayAdLoading´ prop instead and see if it works for you. Also compare your code with the example app.
> Add ´delayAdLoading={1000}´

I have tried it.
`delayAdloading={1000 + id * 1000}`

**Comment by ammarahm-ed** [comment]
Its ´delayAdLoading´ actually, not ´delayAdloading. The documentation has a bug.

**Comment by KVinS** [comment]
> Its ´delayAdLoading´ actually, not ´delayAdloading. The documentation has a bug. @KVinS

Did not help.

**Comment by ammarahm-ed** [comment]
See if you can reproduce the issue in the example since I do not see it in the example app and also update to the latest version 0.3.0 just published today

**Comment by KVinS** [comment]
> See if you can reproduce the issue in the example since I do not see it in the example app and also update to the latest version 0.3.0 just published today
I liked the previous version more...
![2020-07-06_21-10-34](https://user-images.githubusercontent.com/3009807/86625554-e9cdc900-bfcd-11ea-8e9f-e8cfa1c998a7.png)

**Comment by ammarahm-ed** [comment]
@KVinS clone this project and run the example app.

**Comment by ammarahm-ed** [comment]
Try the latest release 0.3.3

**Comment by bhavikgajera92** [comment]
Hi @ammarahm-ed 

I am using this package to show native advanced video ads. In iOS it is showing perfectly fine but in the android It is throwing this kind of error. (Screenshot attached here with.) I am here using the test app Id and test unit Ids provided by Google Ad Mob.

    Android App Id : ca-app-pub-3940256099942544~3347511713
    androidAdUnitCode: 'ca-app-pub-3940256099942544/1044960115',
    iOSAdUnitCode: 'ca-app-pub-3940256099942544/2521693316'

<img width="667" alt="Screenshot 2020-07-13 at 4 45 36 PM" src="https://user-images.githubusercontent.com/5483578/87298778-63167000-c528-11ea-8f44-e38b3605030f.png">

What should be the reason. Please help. I am thankful to you.

**Comment by ammarahm-ed** [comment]
@bhavikgajera92 you are seeing this error because ad is not recieved from the server.   Try to use a vpn app etc or something to be sure your network or phone etc are not blocking it. Also be sure you use Android test Ad Ids to show ads on android. The issue is not related to this lib. Also delete your app from phone. Clear build folders and run again. 


Try the Example app to see if you are seeing the ads or not.

**Comment by bhavikgajera92** [comment]
Hi @ammarahm-ed 
I have checked with VPN also ads are not loading and I tried the Test Ad Ids as well. For the iOS it is working perfectly well for the same internet but for android it is not. Attaching the screenshot for iOS. 
<img width="1008" alt="Screenshot 2020-07-14 at 11 58 19 AM 8" src="https://user-images.githubusercontent.com/5483578/87392224-c4dae680-c5c9-11ea-83c6-1f634fc1c774.png">

Android side still I am getting this.
![image](https://user-images.githubusercontent.com/5483578/87392306-e63bd280-c5c9-11ea-8ce6-c4540c2f0743.png)
I have tested the Ads on same internet on iOS and on Android. On iOS worked perfectly on Android not.
Is there any extra settings to be made at android side ?

Thanks

**Comment by ammarahm-ed** [comment]
No there are no extra settings. The ad should be returned. Try to run the Example App from the project.

**Comment by bhavikgajera92** [comment]
Hi @ammarahm-ed 

Yes I tried to run Example Project and for Native_Ad_Id I am getting the ads. Attaching the screenshot here with. Perfectly fine in Android as well.
<img width="1222" alt="Screenshot 2020-07-14 at 3 55 07 PM" src="https://user-images.githubusercontent.com/5483578/87415333-98839200-c5ea-11ea-93e4-07ffe9b65d63.png">


But for NATIVE_AD_VIDEO_ID I am not able to see any video ads. For iOS I am able to but for android I am not able to see. Attaching the screenshot here with. 
<img width="1222" alt="Screenshot 2020-07-14 at 3 57 22 PM" src="https://user-images.githubusercontent.com/5483578/87415716-20699c00-c5eb-11ea-93a0-a27c5686e592.png">

Thanks

**Comment by ammarahm-ed** [comment]
Server is not returnings ads. Not an issue with this lib.

**Comment by KVinS** [comment]
@bhavikgajera92, Welcome to the club.
Video ads for me only work on a real device.

@ammarahm-ed Hi. Have you added a parameter pass for the [GDPR](https://developers.google.com/admob/android/eu-consent?hl=ru)?

```
Bundle extras = new Bundle();
extras.putString("npa", "1");

AdRequest request = new AdRequest.Builder()
        .addNetworkExtrasBundle(AdMobAdapter.class, extras)
        .build();
```

**Comment by ammarahm-ed** [comment]
Yes added. After getting the consent, pass true or false and rest is handled

**Comment by ammarahm-ed** [comment]
Check the docs. There is a prop to pass consent for EU users.

**Comment by KVinS** [comment]
> Check the docs. There is a prop to pass consent for EU users.
> @KVinS
Thank! In which version did you add requestNonPersonalizedAdsOnly?
You can slightly update the description. Now there is the same California Consumer Privacy Act (CCPA) and The Brazillian LGPD (Lei Geral de Proteção de Dados).

**Comment by ammarahm-ed** [comment]
After **v0.3.1** but update to the latest version to save yourself from bugs in older version.

**Comment by joshuadiezmo** [comment]
I also encounter this problem. any updates on this?

**Comment by bhavikgajera92** [comment]
> @bhavikgajera92, Welcome to the club.
> Video ads for me only work on a real device.
> 
> @ammarahm-ed Hi. Have you added a parameter pass for the [GDPR](https://developers.google.com/admob/android/eu-consent?hl=ru)?
> 
> ```
> Bundle extras = new Bundle();
> extras.putString("npa", "1");
> 
> AdRequest request = new AdRequest.Builder()
>         .addNetworkExtrasBundle(AdMobAdapter.class, extras)
>         .build();
> ```

@KVinS what is this piece of code about ?

**Comment by KVinS** [comment]
> @KVinS what is this piece of code about ?
This is for the [GPDR](https://developers.google.com/admob/android/eu-consent).

**Comment by KVinS** [comment]
@ammarahm-ed
I upgraded to the latest version
(0.3.5)
Now the first ad is showing.
But instead of the second emptiness.
The third is shown again.

**Comment by ammarahm-ed** [comment]
@KVinS  Please share Logcat so we can fix the error

**Comment by FayyazAliKhan1** [comment]
your demo application is not running properly showing this error
_TypeError: undefined is not a function (near '..._$$_REQUIRE(_dependencyMap[3], "react-native-admob-native-ads").AdManager.registerRepository...')
 ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect. 
      This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native._ 
but when I remove all AdManager.registerRepository from index.js the application runs but no ad display

**Comment by ammarahm-ed** [comment]
@FayyazAliKhan1 Example app is working fine.

**Comment by FayyazAliKhan1** [comment]
I mean the build is successful in iOS simulator but this error displays when metro bundler loads completely and the app is showing a white screen

**Comment by FayyazAliKhan1** [comment]
further more I am using my own banner ids and it's almost 4 days that I have implemented it but still ads are not showing to me properly just the above ad logo and small purple box is showing,

**Comment by ammarahm-ed** [comment]
@FayyazAliKhan1 The latest code with repositories is not released yet. Use the master branch instead if you want to try it out.

```
yarn add https://github.com/ammarahm-ed/react-native-admob-native-ads.git
```

You cannot use banner ids with this ad format. You need to generate Native Ad Ids in your Admob Account. Also note that do not use your ad ids in dev mode. Use test ids provided in admob. Read the docs.

**Comment by FayyazAliKhan1** [comment]
@ammarahm-ed I used the master branch but the error is in version 0.5.0 but when I used version of your repo using
`yarn add https://github.com/ammarahm-ed/react-native-admob-native-ads.git` which is version 0.6.0 is working in Android very well Thanks, but this version is failing the build of IOS, and showing the below error, please tell what should I do for IOS.
<img width="573" alt="image" src="https://user-images.githubusercontent.com/46970238/141257609-0676f750-105c-4605-9d1d-c3ac44200d90.png">
